iceoryx_posh/test/moduletests/test_roudi_service_registry.cpp

```diff
@@ -518,52 +531,10 @@ TYPED_TEST(ServiceRegistry_test, AddingMultipleServiceDescriptionAndRemovingAllD
     this->sut.remove(service2);
     this->sut.remove(service3);
 
-    this->sut->find(this->searchResult, iox::capro::IdString_t("a"), iox::capro::Wildcard, iox::capro::Wildcard);
+    this->find(iox::capro::IdString_t("a"), iox::capro::Wildcard, iox::capro::Wildcard);
     EXPECT_THAT(this->searchResult.size(), Eq(0));
 }
 
-TYPED_TEST(ServiceRegistry_test, AddingVariousServiceDescriptionAndGetServicesDoesNotReturnDuplicate)
```

Comment:
I know this is an old test and needs to be reworked. However, I think we still miss a `E` test case from [`ZOMBIES`](https://iceoryx.io/v1.0.1/advanced/best-practice-for-testing/): Exercise Exceptional Behavior. What do you think about creating a complex and random scenario?

Reply:
Not in this PR, this is what we should consider in https://github.com/eclipse-iceoryx/iceoryx/issues/1065 .

Reply:
@mossmaurice Ah, no then we should not add it in the issue (only tests for `ServiceDiscovery`). I would have to think about what this complex scenario would be but I think we have more urgent matters right now. Definitely in this PR and I would not say that something obvious/critical is missing in these test cases right now, so I would create no issue.